src/views/common/Animation/AnimationStatePlaybackBarLayer.tsx

```diff
@@ -1,175 +1,156 @@
+    const { height, dispatch, isPlaying, leftOffset, selectedWindow, updateSelectedWindow } = props
+    const { draw, getBarInterpreter, getEventPoint, barCanvasWidth, getBarClickToFrame, xToFrame, frameToPixelX } = usePlaybackBarGeometry({...props, baseDrawFn: _draw})
+    const { scrubberCenterX, proposalXRange, barInterpreter, barClickToFrame } = usePlaybackBarPixelState({ ...props, getBarInterpreter, getBarClickToFrame, frameToPixelX })
+
+    const { initiateScrubbing, terminateScrubbing, scrubbingStateHandler } = useDraggableScrubber(dispatch, barInterpreter)
+    
+    const handleScrollbarClick = useCallback((x: number, y: number) => _handleScrollbarClick(x, y, barClickToFrame, dispatch), [barClickToFrame, dispatch])
 
-    const handleWheel = useDebouncedWheelHandler(dispatch, isPlaying)
+    const wheelHandler = useWheelHandler(dispatch)
+    const handleWheel = useCallback((e: React.WheelEvent) => !isPlaying && wheelHandler(e), [isPlaying, wheelHandler])
 
-    const getEventPoint = useCallback((e: React.MouseEvent) => {
-        const boundingRect = e.currentTarget.getBoundingClientRect()
-        const point = [e.clientX - boundingRect.x - buttonPanelOffset - leftMargin, e.clientY - boundingRect.y]
-        return point
-    }, [buttonPanelOffset])
+    const dragSelectInitiationRef = useRef<number | undefined>(undefined)
+    const { initiateDragSelection, terminateDragSelection, selectionUpdater } = useDragSelection(updateSelectedWindow, selectedWindow, dispatch, xToFrame)
 
-    const draggingXRef = useRef<number | undefined>(undefined)
-    const wasPlayingRef = useRef<boolean>(false)
     const handleMouseDown = useCallback((e: React.MouseEvent) => {
+        if (e.buttons !== 1) return // we only care about left-clicks
         const [x, y] = getEventPoint(e)
-        if (norm([x - drawData.seenX, barAreaVCenter - y], 2) < scrubberRadius)
-        {
-            // start dragging
-            draggingXRef.current = x
-            // and pause while dragging, if we weren't already playing
-            if (isPlaying) {
-                wasPlayingRef.current = true
-                dispatch({type: 'TOGGLE_PLAYBACK'})
-            }
-        }
-        else {
-            const newFrame = getNewFrame(x, y)
-            if (!newFrame) return
-            dispatch({type: 'SET_CURRENT_FRAME', newIndex: newFrame})
-        }
-    }, [getEventPoint, drawData.seenX, barAreaVCenter, isPlaying, dispatch, getNewFrame])
-
+        if (initiateScrubbing(x, y, isPlaying)) return // if the user grabbed the scrubber, don't do anything else
+        dragSelectInitiationRef.current = window.setTimeout(() => {
+            dragSelectInitiationRef.current = undefined
+            initiateDragSelection(x)
+        }, dragSelectInitiationDelayMs)
+    }, [getEventPoint, isPlaying, initiateScrubbing, initiateDragSelection])
```

Comment:
So the trick with drag-select and also supporting click-selection is that you don't know from a click whether you are starting a drag or just clicking.

To get around this, the way this is handled is when the user clicks, this schedules a callback. If the callback resolves, the state is set to "dragging-selecting." If a mouseup event happens before the callback resolves, the mouseup cancels the callback and treats it as a click interaction (which moves to the clicked-on frame).